CMake/FindSuiteSparse.cmake

```diff
@@ -122,15 +122,15 @@ if(SUITESPARSE_LIBRARY)
   endforeach(mod)
 else()
   if(NOT SUITESPARSE_ROOT_DIR)
-    message(STATUS "Suitesparse dir not found! Please provide correct filepath.")
+    message(FATAL_ERROR "Suitesparse dir not found! Please provide correct filepath.")
     set(SUITESPARSE_DIR ${SUITESPARSE_DIR} CACHE PATH "Path to Suitesparse installation root.")
     unset(SUITESPARSE_LIBRARY CACHE)
     unset(SUITESPARSE_INCLUDE_DIR CACHE)
     unset(SUITESPARSE_LIBRARY_DIR CACHE)
   elseif(NOT SUITESPARSE_LIBRARY)
-    message(STATUS "Suitesparse library not found! Please provide correct filepath.")
+    message(FATAL_ERROR "Suitesparse library not found! Please provide correct filepath.")
   endif()
   if(SUITESPARSE_ROOT_DIR AND NOT SUITESPARSE_INCLUDE_DIR)
-    message(STATUS "Suitesparse include dir not found! Please provide correct filepath.")
+    message(FATAL_ERROR "Suitesparse include dir not found! Please provide correct filepath.")
```

Comment:
Not finding SuiteSparse should not be a fatal error. It is an optional dependency.

Reply:
Suitesparse is only searched for if the related variable is enabled, in which case I think it _should_ be a fatal error.

```cmake
# Root CMakeLists.txt
if(GRIDKIT_ENABLE_SUNDIALS_SPARSE)
  include(FindSuiteSparse)
endif()
```

Reply:
Not really, because one can always fall back to dense or different sparse solver. Logic deciding if the error is fatal or not should imho be outside the find function. The function itself should only report if it succeeded in finding the library or not.